source/opt/decoration_manager.cpp

```diff
@@ -60,39 +60,27 @@ bool DecorationManager::HaveTheSameDecorations(uint32_t id1,
   return true;
 }
 
-// TODO(pierremoreau): Handle SpvOpDecorateId by converting them to a regular
-//                     SpvOpDecorate.
-bool DecorationManager::AreDecorationsTheSame(
-    const ir::Instruction* inst1, const ir::Instruction* inst2) const {
-  //  const auto decorateIdToDecorate = [&constants](const Instruction& inst) {
-  //    std::vector<Operand> operands;
-  //    operands.reserve(inst.NumInOperands());
-  //    for (uint32_t i = 2u; i < inst.NumInOperands(); ++i) {
-  //      const auto& j = constants.find(inst.GetSingleWordInOperand(i));
-  //      if (j == constants.end())
-  //        return Instruction(inst.context());
-  //      const auto operand = j->second->GetOperand(0u);
-  //      operands.emplace_back(operand.type, operand.words);
-  //    }
-  //    return Instruction(inst.context(), SpvOpDecorate, 0u, 0u, operands);
-  //  };
-  //  Instruction tmpA = (deco1.opcode() == SpvOpDecorateId) ?
-  //  decorateIdToDecorate(deco1) : deco1;
-  //  Instruction tmpB = (deco2.opcode() == SpvOpDecorateId) ?
-  //  decorateIdToDecorate(deco2) : deco2;
-  //
-  if (inst1->opcode() == SpvOpDecorateId || inst2->opcode() == SpvOpDecorateId)
-    return false;
+// TODO(pierremoreau): If OpDecorateId is referencing an OpConstant, one could
+//                     check that the constants are the same rather than just
+//                     looking at the constant ID.
+bool DecorationManager::AreDecorationsTheSame(const ir::Instruction* inst1,
+                                              const ir::Instruction* inst2,
+                                              bool ignore_target) const {
```

Comment:
Would it be useful to default ignore_target to 'true'?  Not a big fan of default values, but it seems to be mostly used with this value.

Reply:
Even better in my mind is to avoid a bool param by using two different method names that more explicitly call out the difference.  I find that an accumulation of bool params tends to become hard-to-use since you always have to go back and read the comment to remember what sense the bool has.

But I haven't thought enough to invent good method names for the two cases.  Normally I think of a "decoration" as the the attributes being applied, not including the target.  But here we're using "decoration" to mean "decoration instruction".  So maybe 
SameDecorationValueAndTarget vs. SameDecorationValue  are the two cases?

Reply:
I will leave it for another PR